### PR TITLE
ARROW-104 Improve Error Message for Lib ImportError

### PR DESCRIPTION
**Pull request by blink1073**

### CODE REVIEW

File: bindings/python/pymongoarrow/__init__.py [hunk 1]
```diff
@@ -29,8 +29,10 @@ def _parse_version(version):
 
 try:
     from pymongoarrow.lib import libbson_version
-except ImportError:
-    warnings.warn("This library has not been compiled")
+except ImportError as e:
+    warnings.warn(
+        f"Could not find compiled pymongoarrow.lib extension, please install from source or report the following traceback on the issue tracker:\n{e}"
+    )
```
Comment (ShaneHarvey) [inline bindings/python/pymongoarrow/__init__.py:35]: line length

Comment (ShaneHarvey) [reply bindings/python/pymongoarrow/__init__.py:35]: Should we include the full traceback in the warning?

Comment (blink1073) [reply bindings/python/pymongoarrow/__init__.py:35]: Sounds good

File: bindings/python/pymongoarrow/__init__.py [hunk 2]
```diff
@@ -29,10 +30,12 @@ def _parse_version(version):
...
 try:
     from pymongoarrow.lib import libbson_version
-except ImportError as e:
+except ImportError:
     warnings.warn(
-        f"Could not find compiled pymongoarrow.lib extension, please install from source or report the following traceback on the issue tracker:\n{e}"
+        "Could not find compiled pymongoarrow.lib extension, please install "
+        "from source or report the following traceback on the issue tracker"
     )
+    warnings.warn(traceback.print_exc())  # type: ignore[arg-type]
```
Comment (ShaneHarvey) [inline bindings/python/pymongoarrow/__init__.py:38]: The arg-type is a real bug. This need to use format_exc. Ideally, the format_exc() would be included in the warning above so the user doesn't see two UserWarnings.

Comment (blink1073) [reply bindings/python/pymongoarrow/__init__.py]: ACK

**Review by ShaneHarvey — APPROVED** [review]
LGTM can you provide an example of what the new behavior looks like?

**Comment by blink1073** [comment]
```python
/Users/steve.silvester/workspace/mongo-arrow/bindings/python/pymongoarrow/__init__.py:34: UserWarning: Could not find compiled pymongoarrow.lib extension, please install from source or report the following traceback on the issue tracker:
Traceback (most recent call last):
  File "/Users/steve.silvester/workspace/mongo-arrow/bindings/python/pymongoarrow/__init__.py", line 32, in <module>
    from pymongoarrow.lib import libbson_version
ModuleNotFoundError: No module named 'pymongoarrow.lib'

  warnings.warn(
```

**Pull request merged**